email-common/src/main/java/io/commercetools/sunrise/email/EmailSender.java

```diff
@@ -41,19 +41,43 @@
      * <pre>{@code
      * String messageID = completionStage.toCompletableFuture().join();
      * }</pre>
-     * Exceptions that occur while sending the email are instances of {@link EmailSenderException} that are
-     * wrapped in an unchecked {@link CompletionException} thrown by {@link CompletableFuture#join()}.
+     * <h1>Exception handling</h1>
+     * Exceptions that occur while sending the email are instances of {@link EmailDeliveryException} that are contained
+     * in the {@link CompletionStage} returned by this method. (Note that it is possible that other
+     * {@link RuntimeException}s or {@link Error}s besides {@link EmailDeliveryException} may be contained in the
+     * {@link CompletionStage}.) If {@link CompletableFuture#join()} is invoked like in
+     * above snippet, any {@link EmailDeliveryException} raised while sending the email is wrapped in a
+     * {@link CompletionException} thrown by {@link CompletableFuture#join()}.
+     * <p>
+     * Exceptions that occur while creating an email are instances of {@link EmailCreationException} that are thrown
+     * by {@link #send(MessageEditor)}.
      * <p>
+     * If desired, both kinds of exceptions can be handled like in the following
+     * snippet, without requiring the use {@link CompletableFuture#join()}.
+     * <pre>{@code
+     * try {
+     *      emailSender.send(...)
+     *          .exceptionally(throwable -> {
+     *              if (throwable.getCause() instanceof EmailDeliveryException) {
+     *                  // handle the EmailDeliveryException
+     *              }
+     *              ...
+     *          });
+     * } catch (EmailCreationException e) {
+     *      // handle the EmailCreationException
+     * }
+     * }</pre>
+     * <h1>A note to implementors</h1>
      * Implementations of this method need to apply timeouts that may be configured when creating the {@link EmailSender}
      * instance. The timeouts avoid denial of service by too many connections waiting for stalled I/O.</p>
      *
      * @param messageEditor the email sender passes an empty message to the message editor that the editor shall fill.
      *                      Note that {@link MimeMessage} instances are not immutable. Messages passed to the editor
      *                      must only be used by that editor instance and must not be passed elsewhere.
      * @return A completion stage for sending the message asynchronously.
```

Comment:
I would put in the return:
`@return A completion stage containing a string identifying the delivered email, or an CompletionException with cause {@link EmailDeliveryException} if the email could not be sent successfully`